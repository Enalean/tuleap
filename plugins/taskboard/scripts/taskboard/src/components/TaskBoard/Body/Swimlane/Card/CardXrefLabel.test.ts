/*
 * Copyright (c) Enalean, 2019 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import { shallowMount } from "@vue/test-utils";
import CardXrefLabel from "./CardXrefLabel.vue";
import type { Card } from "../../../../../type";

describe("CardXrefLabel", () => {
    it("displays the xref and the label of a card", () => {
        const wrapper = shallowMount(CardXrefLabel, {
            props: {
                card: {
                    id: 43,
                    label: "Story 2",
                    xref: "story #43",
                    color: "lake-placid-blue",
                    artifact_html_uri: "/path/to/43",
                    is_in_edit_mode: false,
                } as Card,
                label: "Story 2",
            },
        });
        expect(wrapper.element).toMatchSnapshot();
    });

    it("displays the xref and the label using the background color if it exists", () => {
        const wrapper = shallowMount(CardXrefLabel, {
            props: {
                card: {
                    id: 43,
                    label: "Story 2",
                    xref: "story #43",
                    color: "lake-placid-blue",
                    background_color: "fiesta-red",
                    artifact_html_uri: "/path/to/43",
                    is_in_edit_mode: false,
                } as Card,
                label: "Story 2",
            },
        });
        expect(wrapper.get("[data-test=xref]").classes("tlp-swatch-fiesta-red")).toBe(true);
    });

    it("hides the label when card is in edit mode", () => {
        const wrapper = shallowMount(CardXrefLabel, {
            props: {
                card: {
                    id: 43,
                    label: "Story 2",
                    xref: "story #43",
                    color: "lake-placid-blue",
                    background_color: "fiesta-red",
                    artifact_html_uri: "/path/to/43",
                    is_in_edit_mode: true,
                } as Card,
                label: "Story 2",
            },
        });
        expect(wrapper.text()).toBe("story #43");
    });
});
