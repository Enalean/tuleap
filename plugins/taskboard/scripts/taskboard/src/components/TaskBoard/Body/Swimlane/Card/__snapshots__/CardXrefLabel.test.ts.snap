// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardXrefLabel displays the xref and the label of a card 1`] = `
<div
  class="taskboard-card-xref-label"
>
  <a
    class="taskboard-card-xref tlp-swatch-lake-placid-blue"
    data-not-drag-handle="true"
    data-test="xref"
    href="/path/to/43"
  >
    story #43
  </a>
  <span
    class="taskboard-card-label"
  >
    Story 2
  </span>
</div>
`;
